compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9355,20 +9364,32 @@ private BLangBlockStmt getSafeErrorAssignment(Location location, BLangSimpleVarR
                         .anyMatch(retType -> types.isAssignable(errorType, retType)));
 
         String patternFailureCaseVarName = GEN_VAR_PREFIX.value + "t_failure";
-        BLangSimpleVariable errorVar =
-                ASTBuilderUtil.createVariable(location, patternFailureCaseVarName, symTable.errorType,
-                                                createTypeCastExpr(ref, symTable.errorType),
-                                              new BVarSymbol(0, names.fromString(patternFailureCaseVarName),
-                                                             this.env.scope.owner.pkgID, symTable.errorType,
-                                                             this.env.scope.owner, location, VIRTUAL));
+        BVarSymbol errorVarSymbol;
+        if (!isCheckPanicExpr && this.onFailClause != null && invokableSymbol.origin != VIRTUAL) {
```

Comment:
is this this.onFailClause != null needed?

Reply:
It's due to synthetic error variable was created per on-fail clause. But actually we can create the error variable per invokable symbol (functions). Addressed this with https://github.com/ballerina-platform/ballerina-lang/pull/38114/commits/0302205c1d92eaa5218c537c9f5a9d900398cd96